src/ast/mod.rs

```diff
@@ -1189,304 +1123,13 @@ pub type Action = GenericAction<String, String>;
 pub(crate) type MappedAction = GenericAction<CorrespondingVar<String, String>, String>;
 pub(crate) type ResolvedAction = GenericAction<ResolvedCall, ResolvedVar>;
 
-#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```

Comment:
I noticed that the original documentation of GenericAction and GenericFact has been removed. Could you add them back?